programs/bpf/c/sdk/inc/sol_bpf.h

```diff
@@ -99,8 +99,8 @@ typedef struct {
   SolPubkey *key;        /** Public Key of the account owner */
   int64_t *tokens;       /** Numer of tokens owned by this account */
   uint64_t userdata_len; /** Length of userdata in bytes */
-  uint8_t *userdata;     /** On-chain data owned by this account */
-  SolPubkey *program_id; /** Program that owns this account */
+  const uint8_t *userdata;     /** On-chain data owned by this account */
```

Comment:
userdata shouldn’t be const right?

Reply:
I was going for a constant pointer (data can change but pointer cannot but that won't work here for user data).  But this came up because compiler threw an error at line 177:
```
ka[i].userdata = input;
```
Why is input const: `const uint8_t *input`

The user is expected to change the input data.  Are you defending against a user changing the serialized buffer and if so then in `serialize()` you need to cast in order to assign `input` to `userdata`